examples/even-odd-list-with-type-level-booleans

```diff
@@ -0,0 +1,45 @@
+/// A list of type `a` that uses a Boolean at the type level to track
+/// whether its length is even or odd.
+enum EOList[a: Type, even: Bool] {
+    case Nil
+    case Cons(a, EOList[a, not even])
+}
+
+/// Returns an empty list of type `a` with even length (which is 0).
+def mkNil(): EOList[a, true] = EOList.Nil
+
+/// Attaches an element to the front of the list.
+def cons(x: a, l: EOList[a, e]): EOList[a, not e] =
+    EOList.Cons(x, l)
+
+/// Returns the head of the list.
```

Comment:
Returns the head element of an odd list (since such a list is guaranteed to be empty).

Reply:
I do not know what this means - an odd list is guaranteed to be empty? Wouldn't it be guaranteed not to be empty?

Reply:
I meant non-empty.

Reply:
So we don't support calling `head` for any even lists?

Reply:
That's right

Reply:
(one would have a different function returning an option, but we will not bother with that.)